ingestion/src/metadata/profiler/source/fetcher/fetcher_strategy.py

```diff
@@ -0,0 +1,298 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Entity Fetcher Strategy
+"""
+
+import traceback
+from abc import ABC, abstractmethod
+from typing import Iterable, Iterator, Optional, cast
+
+from metadata.generated.schema.entity.data.database import Database
+from metadata.generated.schema.entity.services.ingestionPipelines.status import (
+    StackTraceError,
+)
+from metadata.generated.schema.metadataIngestion.databaseServiceProfilerPipeline import (
+    DatabaseServiceProfilerPipeline,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    OpenMetadataWorkflowConfig,
+)
+from metadata.generated.schema.settings.settings import Settings
+from metadata.ingestion.api.models import Either
+from metadata.ingestion.api.status import Status
+from metadata.ingestion.models.entity_interface import EntityInterfaceWithTags
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.profiler.source.fetcher.profiler_source_factory import (
+    profiler_source_factory,
+)
+from metadata.profiler.source.model import ProfilerSourceAndEntity
+from metadata.utils.db_utils import Table
+from metadata.utils.filters import (
+    filter_by_classification,
+    filter_by_database,
+    filter_by_schema,
+    filter_by_table,
+)
+
+FIELDS = ["tableProfilerConfig", "columns", "customMetrics", "tags"]
+
+
+class FetcherStrategy(ABC):
+    """Fetcher strategy interface"""
```

Comment:
Can you add a bit more information about this class? When would we want to use it and in what context?